packages/terra-menu/src/MenuItem.jsx

```diff
@@ -359,8 +367,8 @@ class MenuItem extends React.Component {
             {...attributes}
             className={classNames(itemClassNames, cx(theme.className))}
             ref={this.setItemNode}
-            role={role}
-            aria-checked={markAsToggled}
+            role={menuRole}
+            {...selectAttribute}
```

Comment:
Seems like you could just create an aria-selected attribute with the necessary condition here.  Do we need a spread operator for a single attribute?

Reply:
1. In terra-menu we were using three roles such as 'menuitem' , 'menuitemradio' and 'menuitemcheckbox', and we were using 'aria-selected' attribute as well.
2. 'aria-selected' attribute should be used when role was 'menuitemradio' and 'menuitemcheckbox', if use 'aria-selected' attribute for role='menuitem' it will throw accessibility violations error (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-checked#associated_roles ), to avoid this we are storing it and binding using spread operator.

Reply:
If you provide null or undefined for a value, the attribute will not be rendered.  It still seems like you could simplify the logic for this attribute and not need a spread operator.

Reply:
Updated condition without spread operator - [10f62c6](https://github.com/cerner/terra-framework/pull/1718/commits/10f62c6bde740a1fc034c667e1403352ff3b83b2)